### PR TITLE
Run tests in iPhone simulator

### PR DESCRIPTION
**Pull request by PavelPunegov**
- iPhone simulator support
- Refactor QEMU, WASM and SSH execution

### CODE REVIEW

File: buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/ExecutorService.kt [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                    val qemu = if (platform.target === KonanTarget.LINUX_MIPS32) "qemu-mips" else "qemu-mipsel"
+                    val absoluteQemu = "$absoluteTargetToolchain/bin/$qemu"
+                    val exe = executable
+                    executable = absoluteQemu
+                    args = listOf("-L", absoluteTargetSysRoot, exe) + args
+                }
+            }
+        }
+
+        KonanTarget.IOS_X64 -> ExecSimulator(project)
```
Comment (olonho) [inline buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/ExecutorService.kt]: New instance each time vs. singleton in other branches?

Comment (PavelPunegov) [reply buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/ExecutorService.kt]: Well, an executor is created once for a test target platform. It's no need to create a singlton

File: buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/ExecutorService.kt [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    override fun execute(action: Action<in ExecSpec>): ExecResult? = project.exec { execSpec ->
+        action.execute(execSpec)
+        with(execSpec) { commandLine = listOf(simctl, "spawn", iosDevice, executable) + args }
+    }
+}
+
+/**
+ * Remote process executor.
+ *
+ * @property remote=user@host makes binaries be executed on a remote host
```
Comment (alexander-gorshenev) [inline buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/ExecutorService.kt]: -Premote=...

File: buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/ExecutorService.kt [hunk 3]
```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
```
Comment (alexander-gorshenev) [inline buildSrc/plugins/src/main/kotlin/org/jetbrains/kotlin/ExecutorService.kt]: 2018

**Pull request merged**